### PR TITLE
Feat: table design changes

### PR DESCRIPTION
**Pull request by danad02**
- Update `Tooltip` component so that parent element can use `overflow: hidden`.
- Update `Table` design.
- Update `Table` column width with more sane values - resizing should now feel more comfortable.

Linked to #306 and #273

### CODE REVIEW

File: packages/ui/__tests__/Table/Table.test.tsx [hunk 1]
```diff
@@ -168,7 +138,7 @@ describe('Table', () => {
       numberOfItems: smallDataSet.length,
     })
 
-    expect(wrapper.findDataTest('table-noData-cell').exists()).toBe(false)
+    expect(wrapper.find(EmptyState).exists()).toBe(false)
```
Comment (Sodik) [inline packages/ui/__tests__/Table/Table.test.tsx:141]: I suggest passing test id into EmptyState. This might be very useful for e2e tests

**Review by Sodik — CHANGES_REQUESTED** [review]

File: packages/ui/src/Table/Table.tsx [hunk 2]
```diff
@@ -172,7 +174,9 @@ type CustomTableProps<D extends object> = {
   paginationClassName?: string
   footerClassName?: string
   contentClassName?: string
-  noDataTitle?: string | ReactNode
+  noDataTitle?: string
```
[the remark this answers is not in the document]
Comment (danad02) [reply packages/ui/src/Table/Table.tsx:177]: You are correct. I wanted to revert this change but forgot :+1:

[the remark this answers is not in the document]
Comment (danad02) [reply packages/ui/src/Table/Table.tsx]: Changes in https://github.com/hazelcast/hive/pull/314/commits/12a8332dd23235cd3260ce320326a2a30d3285fc

File: packages/ui/__tests__/Table/Table.test.tsx (hunk 1 above)
Comment (danad02) [reply packages/ui/__tests__/Table/Table.test.tsx]: Changed in https://github.com/hazelcast/hive/pull/314/commits/12a8332dd23235cd3260ce320326a2a30d3285fc

**Review by Sodik — APPROVED** [review]

**Pull request merged**